src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -57,7 +57,11 @@ namespace impl {
  *       detail, and, normally, should not be used by customer apps
  */
 template <bool Serialize, typename TeamMember, typename Lambda, typename ValueType>
+#ifdef KOKKOS_ENABLE_SYCL
+KOKKOS_INLINE_FUNCTION SYCL_EXTERNAL
```

Comment:
We do have lots of KOKKOS_INLINE_FUNCTION in EKAT. Why does this particular case require a special handling? And what does this macro do differently from KOKKOS_INLINE_FUNCTION?

Reply:
There is a bit of limitation on how static device-side functions and linking can be setup with SYCL. There are majorly two changes suggested for this function alone and has nothing to do with `KOKKOS_INLINE_FUNCTION`. 

Regarding the need of `SYCL_EXTERNAL`: In the case of linking C++ functions to a SYCL application, where the definitions are not available in the same translation unit of the compiler, then the macro SYCL_EXTERNAL has to be provided. More info [here](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:syclexternal)

I have made some changes to cleanup this. Thanks.

Reply:
Ah, I misread the two macros as just a single one.

Reply:
But still, why do we need external here? The `parallel_reduce` fcn is defined inline here, so it should be available in all translation units. Am I missing something?

Reply:
But then we should not need `SYCL_EXTERNAL` at all, right?

Reply:
Making the function `static` was being problematic to device-side kernel. `SYCL_EXTERNAL` attr and removing `static` was suggested. I have slightly modified the function signature to look as below, please let me know if that looks okay
 
```c++
template <bool Serialize, typename TeamMember, typename Lambda, typename ValueType>
#ifdef KOKKOS_ENABLE_SYCL
SYCL_EXTERNAL
#else
static
#endif
KOKKOS_INLINE_FUNCTION
void parallel_reduce (const TeamMember& team,
                            const int& begin, // pack index                                                                                                                                
                            const int& end, // pack index                                                                                                                                  
                            const Lambda& lambda,
                            ValueType& result)
```

Reply:
Are you saying that SYCL complains about the function being static, and requires the external keyword to compile? Either I am not understanding SYCL, or this seems wrong. The "inline" and "extern" keywords are somewhat mutually exclusive, so I am surprised that SYCL needs that macro. The documentation suggests that it is needed if the definition is not available to other TU's, but since it's inlined, it should be available as soon as this header is included.

Anyhow, if you say that it does not compile without that macro, then it's fine. I'm just not really happy with this confusing detail.

Reply:
Sorry for the confusion. The need for SYCL_EXTERNAL is not required any more. There has been a mix-up with compiler versions at my end. Thanks for pointing this out.